src/plugins/visualizations/public/services.ts

```diff
@@ -64,5 +65,17 @@ export const [getOverlays, setOverlays] = createGetterSetter<OverlayStart>('Over
 
 export const [getChrome, setChrome] = createGetterSetter<ChromeStart>('Chrome');
 
-export const [getSavedSearchLoader, setSavedSearchLoader] =
-  createGetterSetter<SavedObjectLoader>('savedSearchLoader');
+const [_getSpaces, setSpaces] = createGetterSetter<SpacesPluginStart>('Spaces');
+
+/** "spaces" is optional, in case if that plugin is turned off "_getSpaces" method trigger the exception  **/
+const getSpaces = () => {
+  let spaces: SpacesPluginStart | undefined;
+  try {
+    spaces = _getSpaces();
+  } catch {
+    // nothing to be here
+  }
+  return spaces;
+};
+
+export { getSpaces, setSpaces };
```

Comment:
Since the `createGetterSetter` method does not much besides being a getter and setter that throws if the value is undefined (which we ignore here), I wonder if it wouldn't be nicer code, if we simply have those methods inline here:

```ts
let spaces: SpacesPluginStart | undefined;
export setSpaces = (s) => spaces = s;
export getSpaces = () => spaces;
```

Reply:
@timroes It's a common issue, I think instead of your suggestion we can enhance a little `createGetterSetter` and pass one more argument. 

`export const createGetterSetter = <T extends object>(name: string, isValueRequired: boolean = true): [Get<T>, Set<T>] => { `

It will be better, cause then we can easily update all similar cases. I'm in a middle of preparing PR for that....

Reply:
#114128